domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Testcontainers/WalletContainer.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using DotNet.Testcontainers.Builders;
+using DotNet.Testcontainers.Containers;
+using Testcontainers.PostgreSql;
+using Xunit;
+
+namespace API.IntegrationTests.Testcontainers;
+
+public class WalletContainer : IAsyncLifetime
+{
+    private readonly Lazy<IContainer> walletContainer;
+    private readonly PostgreSqlContainer postgresContainer;
+
+    public WalletContainer()
+    {
+        postgresContainer = new PostgreSqlBuilder()
+            .WithDatabase("postgres")
+            .WithUsername("postgres")
+            .WithPassword("postgres")
+            .WithExposedPort(5432)
+            .WithPortBinding(5432, true)
+            .Build();
+
+        walletContainer = new Lazy<IContainer>(() =>
+        {
+            var postgresConnectionString = $"Host={postgresContainer.IpAddress};Port=5432;Database=postgres;Username=postgres;Password=postgres";
+
+            return new ContainerBuilder()
+                .WithImage("ghcr.io/project-origin/wallet-server:0.1.3")
+                .WithPortBinding(80, true)
+                .WithCommand("--serve", "--migrate")
+                .WithEnvironment("ConnectionStrings__Database", postgresConnectionString)
+                .WithEnvironment("ServiceOptions__EndpointAddress", $"http://postgres:5432/")
```

Comment:
This is probably not an issue wrt. the current usage of the Wallet by the TransferAgreement domain, but this is not the correct setting for ServiceOptions:EndpointAddress. It needs to be the endpoint reachable on the host, which requires that you know the host port assigned to the container. For this reason we have hardcoded the host port in the certificates domain to 7890, which means that the endpoint address becomes "http://localhost:7890".

Reply:
See https://github.com/Energinet-DataHub/energy-origin/blob/main/domains/certificates/Query.API/API.IntegrationTests/Testcontainers/ProjectOriginStack.cs